processor/cumulativetodeltaprocessor/internal/tracking/tracker.go

```diff
@@ -159,7 +160,7 @@ func (t *MetricTracker) Convert(in MetricPoint) (out DeltaValue, valid bool) {
 			value.Sum = prevValue.Sum
 		}
 
-		if len(value.Buckets) != len(prevValue.Buckets) {
+		if len(value.BucketCounts) != len(prevValue.BucketCounts) || !slices.Equal(value.BucketBounds, prevValue.BucketBounds) {
```

Comment:
To the point you made in the PR description:

>I am comparing the bucket bounds (floats) by equality; would we expect SDKs to sometimes send slightly different bucket bounds? The specification seems to say very little about the stability of histogram bucket bounds unfortunately.

I don't have a good answer for this either, but this is a good callout, and I think it's safe to say this is a possibility. Even regardless of what the OTel spec may say, we may get input values from other metric formats that may have looser requirements. Could we emit a log when we hit this case so we can track this?

I think the alternative would be to implement something like the `(*ExponentialBuckets).Coarsen` method that tries to stabilize the bucket boundaries.